Dockerfiles/manifests/openshift/tf-cluster/README.md

```diff
@@ -0,0 +1,352 @@
+# terraform-aws-openshift
+
+[![CircleCI](https://circleci.com/gh/dwmkerr/terraform-aws-openshift.svg?style=shield)](https://circleci.com/gh/dwmkerr/terraform-aws-openshift)
+
+
+**Index**
+
+<!-- TOC depthFrom:2 depthTo:6 withLinks:1 updateOnSave:1 orderedList:0 -->
+
+- [Overview](#overview)
+- [Prerequisites](#prerequisites)
+- [Creating the Cluster](#creating-the-cluster)
+- [Installing OpenShift](#installing-openshift)
+- [Accessing and Managing OpenShift](#accessing-and-managing-openshift)
+	- [OpenShift Web Console](#openshift-web-console)
+	- [The Master Node](#the-master-node)
+	- [The OpenShift Client](#the-openshift-client)
+- [Connecting to the Docker Registry](#connecting-to-the-docker-registry)
+- [Additional Configuration](#additional-configuration)
+- [Choosing the OpenShift Version](#choosing-the-openshift-version)
+- [Destroying the Cluster](#destroying-the-cluster)
+- [Makefile Commands](#makefile-commands)
+- [Pricing](#pricing)
+- [Troubleshooting](#troubleshooting)
+- [Developer Guide](#developer-guide)
+	- [CI](#ci)
+	- [Linting](#linting)
+
+
+<!-- /TOC -->
+
+## Overview
+
+Terraform is used to create infrastructure as shown:
+
+![Network Diagram](./docs/network-diagram.png)
+
+Once the infrastructure is set up an inventory of the system is dynamically
+created, which is used to install the OpenShift Origin platform on the hosts.
+
+## Prerequisites
+
+You need:
+
+1. [Terraform](https://www.terraform.io/intro/getting-started/install.html) - `brew update && brew install terraform`
+2. An AWS account, configured with the cli locally -
+```
+if [[ "$unamestr" == 'Linux' ]]; then
+        dnf install -y awscli || yum install -y awscli
+elif [[ "$unamestr" == 'FreeBSD' ]]; then
+        brew install -y awscli
+fi
+```
+3. Export the follwing environment variables -
+```
+export AWS_ACCESS_KEY_ID=
+export AWS_SECRET_ACCESS_KEY=
+export TF_VAR_AWS_SECRET_ACCESS_KEY=...
+export TF_VAR_AWS_ACCESS_KEY_ID=...
+export TF_VAR_AWS_REGION=us-east-1
+```
+
+## Creating the Cluster
+
+Create the infrastructure first:
+
+```bash
+# Make sure ssh agent is on, you'll need it later.
+eval `ssh-agent -s`
+
+# Create the infrastructure.
+make infrastructure
+```
+
+You will be asked for a region to deploy in, use `us-east-1` or your preferred region. You can configure the nuances of how the cluster is created in the [`main.tf`](./main.tf) file. Once created, you will see a message like:
+
+```
+$ make infrastructure
+var.region
```

Comment:
if we provide `TF_VAR_AWS_REGION` do you still get prompted for a region ?
otherwise lets remove this section